src/internal/m365/collection/site/lists.go

```diff
@@ -21,13 +21,19 @@ func ListToSPInfo(lst models.Listable) *details.SharePointInfo {
 		count    = len(lst.GetItems())
 	)
 
+	template := ""
+	if lst.GetList() != nil {
+		template = ptr.Val(lst.GetList().GetTemplate())
+	}
+
 	return &details.SharePointInfo{
-		ItemType:  details.SharePointList,
-		ItemName:  name,
-		ItemCount: int64(count),
-		Created:   created,
-		Modified:  modified,
-		WebURL:    webURL,
+		ItemType:     details.SharePointList,
+		ItemName:     name,
+		ItemCount:    int64(count),
+		ItemTemplate: template,
```

Comment:
I'm a little confused by the value naming here.  The list has N (0 or more) items.  Those are counted in `ItemCount`.  But we have three other properties: `ItemType, ItemName, ItemTemplate` with only one value.  I think that the `Item`, in this case, is supposed to reference the list itself.  But that's a conflict of naming, because the list is not an item, and it's going to cause confusion.  Not only will it be confusing when trying to figure out whether an `Item` refers to a list or not, it will also be difficult to know if other types (ie: libraries) are supposed to populate ItemTemplate and ItemCount and etc.

Since we're only introducing support now, I think it's the right time to move this data to a substruct within sharePointInfo like:

```go
type details.SharePointInfo{
  List            *ListInfo
  ItemType   enum
}

type ListInfo struct {
  Name           string
  ItemCount   int64
  Template     string
  Created       time.TIme
  WebURL      string
  Modified      time.Time
  ...
}
```

This will minimize cross-contamination and ensure we have clear and identifiable properties specifically for list items themselves.  It'll have some structural overlap with other SharePointInfo properties, and that's fine.  Better to be duplicated and easy to understand than to be confusing.

Reply:
Yes, here the Item represent the `list` itself.
Yes, this contract is much better. Initially I thought about the same, just like they way in `Conversation` and `Channels` in GroupsInfo.
But I got a bit influenced by `ExchangeInfo` having `Events`, `Contacts`, `Mail` fields in same struct. But their fields are quite distinctive to the categories.
I shall make the changes accordingly.